lib/trip_dispatcher.rb

```diff
@@ -1,26 +1,60 @@
-require 'csv'
-require 'time'
+require "csv"
+require "time"
 
-require_relative 'passenger'
-require_relative 'trip'
+require_relative "passenger"
+require_relative "trip"
+require_relative "driver"
 
 module RideShare
   class TripDispatcher
-    attr_reader :drivers, :passengers, :trips
+    attr_accessor :drivers, :passengers, :trips
 
-    def initialize(directory: './support')
+    def initialize(directory: "./support")
       @passengers = Passenger.load_all(directory: directory)
       @trips = Trip.load_all(directory: directory)
+      @drivers = Driver.load_all(directory: directory)
       connect_trips
     end
 
     def find_passenger(id)
       Passenger.validate_id(id)
-      return @passengers.find { |passenger| passenger.id == id }
+      passenger = @passengers.find { |passenger| passenger.id == id }
+      return passenger
+    end
+
+    def find_driver(id)
+      Driver.validate_id(id)
+      return @drivers.find { |driver| driver.id == id }
+    end
+
+    def request_trip(passenger_id)
+      available_driver = @drivers.find { |driver| driver.status == :AVAILABLE }
+
+      raise ArgumentError.new("There are no drivers available") if available_driver == nil
```

Comment:
This should be a [custom error](https://github.com/Ada-Developers-Academy/textbook-curriculum/blob/master/02-intermediate-ruby/custom-exceptions.md), not an argument error. 

There's nothing wrong with the argument itself (changing the `passenger_id` won't help).

For example you could create an error class named `NoDriverAvailableError`:
```suggestion
      raise NoDriverAvailableError.new("There are no drivers available") if available_driver == nil
```